src/services/rest/RestService.ts

```diff
@@ -0,0 +1,148 @@
+      jsonPayload: undefined,
+      cache: cacheTarget,
+      urlSuffix: "/".concat(key)
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("get", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCacheDelete(cacheTarget: string): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: undefined,
+      cache: cacheTarget,
+      urlSuffix: undefined
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("delete", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCacheDeleteKey(cacheTarget: string, key: string): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: undefined,
+      cache: cacheTarget,
+      urlSuffix: "/".concat(key)
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("delete", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCachePut(cacheTarget: string, key: string, data: any): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: data,
+      cache: cacheTarget,
+      urlSuffix: "/".concat(key)
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("put", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCachePost(cacheTarget: string, key: string, data: any): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: data,
+      cache: cacheTarget,
+      urlSuffix: "/".concat(key)
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("post", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCacheQuery(cacheTarget: string, query: string): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: {query: query},
+      cache: cacheTarget,
+      urlSuffix: "?action=search"
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("post", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  clearGetCache(): void {
+    this.$cacheFactory.get("$http").removeAll();
+  }
+
+  private executeHelper(type: string, url: string, data: any, headers?: ng.IRequestShortcutConfig): ng.IPromise<any> {
+    let deferred: ng.IDeferred<any> = this.$q.defer<any>();
+    let response: IHttpPromise;
+    switch (type) {
+      case "post":
+        response = this.$http.post(url, data, headers);
+        break;
+      case "put":
+        response = this.$http.put(url, data, headers);
+        break;
+      case "get":
+        response = this.$http.get(url, headers);
+        break;
+      case "delete":
+        response = this.$http.delete(url, headers);
+        break;
+      default:
+        deferred.reject("Unknown request type " + type);
+    }
+    response.then(
+      (success: any) => {
+        deferred.resolve(success);
+      },
+      (failure: any) => {
+        this.processResponseFailure(deferred, failure);
+      });
```

Comment:
Can be a one liner: `response.then(success => deferred.resolve(success), failure => this.processResponseFailure(deferred, failure));`

Reply:
In fact, `processResponseFailure` is only used here, so might as well do inline.

Reply:
@vblagoje This still needs doing.